redfish-core/lib/service_root.hpp

```diff
@@ -37,21 +37,68 @@ namespace redfish
 inline void
     handleACFWindowActive(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
 {
-    sdbusplus::asio::getProperty<bool>(
-        *crow::connections::systemBus, "com.ibm.PanelApp", "/com/ibm/panel_app",
-        "com.ibm.panel", "ACFWindowActive",
+    // Redfish property ACFWindowActive=true when either of these is true:
+    //  - D-Bus property allow_unauth_upload.  (This is aka the Redfish
+    //    property AllowUnauthACFUpload which the BMC admin can control.)
+    //  - D-Bus property ACFWindowActive.  (This is aka the Redfish
+    //    property ACFWindowActive under /redfish/v1/AccountService/
+    //    Accounts/service property Oem.IBM.ACF.  The value is provided by
+    //    the PanelApp and is true when panel function 74 is active.)
+    // Check D-Bus property allow_unauth_upload first.
+    crow::connections::systemBus->async_method_call(
         [asyncResp](const boost::system::error_code ec,
-                    const bool isACFWindowActive) {
+                    const std::variant<bool>& allowed) {
         if (ec)
         {
-            BMCWEB_LOG_ERROR << "Failed to read ACFWindowActive property";
-            // Default value when panel app is unreachable.
-            asyncResp->res.jsonValue["Oem"]["IBM"]["ACFWindowActive"] = false;
+            BMCWEB_LOG_ERROR
+                << "D-Bus response error reading allow_unauth_upload: " << ec;
+            messages::internalError(asyncResp->res);
             return;
         }
-        asyncResp->res.jsonValue["Oem"]["IBM"]["ACFWindowActive"] =
-            isACFWindowActive;
-        });
+        const bool* allowUnauthACFUpload = std::get_if<bool>(&allowed);
+        if (allowUnauthACFUpload == nullptr)
+        {
+            BMCWEB_LOG_ERROR << "nullptr for allow_unauth_upload";
+            messages::internalError(asyncResp->res);
+            return;
+        }
+        if (*allowUnauthACFUpload)
+        {
+            asyncResp->res.jsonValue["Oem"]["IBM"]["ACFWindowActive"] =
+                *allowUnauthACFUpload;
+            return;
+        }
+
+        // Check D-Bus property ACFWindowActive
+        crow::connections::systemBus->async_method_call(
+            [asyncResp](const boost::system::error_code ec1,
+                        const std::variant<bool>& retVal) {
```

Comment:
include what you use
#include <variant>

Reply:
variant is no longer used after changing to use getProperty<>().